WalkingApp/model.py

```diff
@@ -25,29 +25,37 @@ def get_recent_walks():
     Return the last 4 walks from the database as a list of tuples (date, kms)
     """
     with _open_database() as conn:
-        return [tuple(row) for row in conn.execute("SELECT date, kms FROM WALKS ORDER BY date DESC LIMIT 4")]
+        return [tuple(row) for row in conn.execute("SELECT day, kms FROM WALKS ORDER BY day DESC LIMIT 4")]
 
 
-def save_to_database(date, kms: float, time, kcal: int, steps: int):
+def save_to_database(day, kms: float, time, kcal: int, steps: int):
     """
     Insert a new walk into the database.
-    :param date: date value in the "%d/%m/%y" format
+    :param day: date value in the "%d/%m/%y" format
     :param kms: kilometres
     :param time: duration expressed as the number of minutes
     :param kcal: amount of kcal
     :param steps: amount of steps
     """
     with _open_database() as conn:
-        conn.execute("INSERT INTO WALKS VALUES (?, ?, ?, ?, ?)", [date, kms, time, kcal, steps])
+        conn.execute("INSERT INTO WALKS VALUES (?, ?, ?, ?, ?)", [day, kms, time, kcal, steps])
         conn.commit()
 
 
-def calculate_statistics():
+def calculate_overall_statistics():
     """
     Return the sum of kms, duration, kcal and steps from the database as a tuple.
     """
     with _open_database() as conn:
         return tuple(conn.execute("SELECT SUM(kms), SUM(duration), SUM(kcal), SUM(steps) FROM WALKS").fetchone())
 
 
-selected_date = None
+def calculate_monthly_statistics():
+    """
+    Return the monthly sum of kms, duration, kcal and steps from the database as a tuple.
+    """
+    with _open_database() as conn:
+        return tuple(conn.execute("""SELECT day, CURRENT_DATE, SUM(kms), SUM(duration), SUM(kcal), SUM(steps) FROM WALKS 
+                                     WHERE SUBSTR(day, 4, 2) = SUBSTR(CURRENT_DATE, 6, 2)
+                                     AND SUBSTR(day, 7, 2) = SUBSTR(CURRENT_DATE, 3,2)""").fetchone())
```

Comment:
It is not clear what is going on here. The confusing bit is that `day` is stored as `DD/MM/YY` whereas `CURRENT_DATE` seems to return ISO 8601 date `YYYY-MM-DD`. It would make sense to change the format of the `day` column to match ISO 8601 as well. And then it should be possible to use `strftime` to extract the year and month, which would make the intent of the query clear. Something like `WHERE strftime("%Y-%m", "now") = strftime("%Y-%m", day)`

Reply:
Thank you, I will definitely check it out